src/value.c

```diff
@@ -450,33 +446,17 @@ int SIValue_Compare(const SIValue a, const SIValue b) {
 		return SAFE_COMPARISON_RESULT(diff);
 	}
 
-	return DISJOINT;
+	return a.type - b.type;
 }
 
-// Return a strcmp-like value wherein -1 indicates that the value
-// on the left-hand side is lesser, and 1 indicates that is greater.
+/* Return a strcmp-like value wherein negative result indicates that the value
+ * on the left-hand side is lesser, and 1 indicates that is greater. */
 int SIValue_Order(const SIValue a, const SIValue b) {
-	// If the values are directly comparable, return the comparison result
+	// If the values are directly comparable, return the comparison result.
 	int cmp = SIValue_Compare(a, b);
-	if(cmp != DISJOINT && cmp != COMPARED_NULL) return cmp;
-
-	// Cypher's orderability property defines string < boolean < numeric < NULL.
-	if(a.type == T_STRING) {
-		return -1;
-	} else if(b.type == T_STRING) {
-		return 1;
-	} else if(a.type == T_BOOL) {
-		return -1;
-	} else if(b.type == T_BOOL) {
-		return 1;
-	} else if(a.type & SI_NUMERIC) {
-		return -1;
-	} else if(b.type & SI_NUMERIC) {
-		return 1;
-	}
-
-	// We can reach here if both values are NULL, in which case no order is imposed.
-	return 0;
+	if(cmp != COMPARED_NULL)  return cmp;
+	// Else, return the type diff
+	return a.type - b.type;
```

Comment:
`a.type - b.type` is the result of `SIValue_Compare` where `a` and `b` types differ.
I believe this line is redundant.